codecparsers/h264Parser.cpp

```diff
@@ -107,29 +140,31 @@ PPS::~PPS()
 }
 
 //according to G.7.3.1.1
-void NalUnit::parseSvcExtension(BitReader& br)
+bool NalUnit::parseSvcExtension(BitReader& br)
 {
-    m_svc.idr_flag = br.read(1);
-    m_svc.priority_id = br.read(6);
-    m_svc.no_inter_layer_pred_flag = br.read(1);
-    m_svc.dependency_id = br.read(3);
-    m_svc.quality_id = br.read(4);
-    m_svc.temporal_id = br.read(3);
-    m_svc.use_ref_base_pic_flag = br.read(1);
-    m_svc.discardable_flag = br.read(1);
-    m_svc.output_flag = br.read(1);
-    m_svc.reserved_three_2bits = br.read(2);
+    READ(m_svc.idr_flag);
+    READ_BITS(m_svc.priority_id, 6);
+    READ(m_svc.no_inter_layer_pred_flag);
+    READ_BITS(m_svc.dependency_id, 3);
+    READ_BITS(m_svc.quality_id, 4);
+    READ_BITS(m_svc.temporal_id, 3);
+    READ(m_svc.use_ref_base_pic_flag);
+    READ(m_svc.discardable_flag);
+    READ(m_svc.output_flag);
+    READ_BITS(m_svc.reserved_three_2bits, 2);
```

Comment:
with a inline function this can be changed to return read() || read() || read() || read_bits(), etc.